src/DIRAC/Resources/Computing/AREXComputingElement.py

```diff
@@ -0,0 +1,644 @@
+            if not result["OK"]:
+                self.log.debug("CE %s still not properly initialized. Aborting job submission." % self.ceHost)
+                return S_ERROR("REST interface not initialised.")
+        self.s.cert = os.environ["X509_USER_PROXY"]
+
+        # Will this ever work? Has this ever worked / been tested? See the getJobStatus function below?
+        self.log.debug("Killing jobs", "%s" % jobIDList)
+        jList = []
+        for job in jobIDList:
+            jList.append(self._pilot_toREST(job))
+
+        # List of jobs in json format for the REST query
+        jj = {}
+        jj["job"] = []
+        for job in jList:
+            jj["job"].append({"id": job})
+
+        command = "jobs"
+        params = {"action": "kill"}
+        query = self.base_url + command
+        # Killing jobs should be fast - bulk timeout of 10 seconds should be okay(?)
+        r = self.s.post(query, data=jj, headers=self.headers, params=params, timeout=10.0)
+        if r.ok:
+            # Job successfully submitted
+            self.log.debug("Successfully deleted jobs %s " % (json.loads(r.text.replace("\n", ""))))
+        else:
+            return S_ERROR("Failed to kill all these jobs: %s" % json.loads(r.text.replace("\n", "")))
+
+        return S_OK()
+
+    #############################################################################
+    def getCEStatus(self):
+        """Method to return information on running and pending jobs.
+        We hope to satisfy both instances that use robot proxies and those which use proper configurations.
+        """
+
+        result = self._prepareProxy()
+        if not result["OK"]:
+            self.log.error("AREXComputingElement: failed to set up proxy", result["Message"])
+            return result
+
+        if not self.rest:  # Something went wrong in the initialisation. Redo it.
+            result = self._configureRestInterface()
+            if not result["OK"]:
+                self.log.debug("CE %s still not properly initialized. Cannot get CE status." % self.ceHost)
+                return S_ERROR("REST interface not initialised.")
+        self.s.cert = os.environ["X509_USER_PROXY"]
+
+        # Try to find out which VO we are running for. Essential now for REST interface.
+        vo = ""
+        res = getVOfromProxyGroup()
+        if res["OK"]:
+            vo = res["Value"]
+
+        result = S_OK()
+        result["SubmittedJobs"] = 0
+        command = "info"
+        params = {"schema": "glue2"}
+        query = self.base_url + command
+        r = self.s.get(query, headers=self.headers, params=params, stream=True, timeout=1.0)
+        r.raw.decode_content = True  # Needed for "lots" of information (like the stream argument above)?
+
+        if not r.ok:
+            res = S_ERROR("Unknown failure for CE %s. Is the CE down?" % self.ceHost)
+            return res
+
+        p = json.loads(r.text.replace("\n", ""))
```

Comment:
It should not

```python
In [4]: withSpace = '{}\n\n\n\n'

In [5]: json.loads(withSpace)
Out[5]: {}
```

Reply:
The only way I can imagine getting that error is if you have invalid JSON like:

```python
In [1]: import json
   ...: json.loads('{"job_name": "hello\nworld"}')
---------------------------------------------------------------------------
JSONDecodeError                           Traceback (most recent call last)
<ipython-input-1-4ada5a6af7b6> in <module>
      1 import json
----> 2 json.loads('{"job_name": "hello\nworld"}')

~/miniconda3/envs/dirac-development/lib/python3.9/json/__init__.py in loads(s, cls, object_hook, parse_float, parse_int, parse_constant, object_pairs_hook, **kw)
    344             parse_int is None and parse_float is None and
    345             parse_constant is None and object_pairs_hook is None and not kw):
--> 346         return _default_decoder.decode(s)
    347     if cls is None:
    348         cls = JSONDecoder

~/miniconda3/envs/dirac-development/lib/python3.9/json/decoder.py in decode(self, s, _w)
    335
    336         """
--> 337         obj, end = self.raw_decode(s, idx=_w(s, 0).end())
    338         end = _w(s, end).end()
    339         if end != len(s):

~/miniconda3/envs/dirac-development/lib/python3.9/json/decoder.py in raw_decode(self, s, idx)
    351         """
    352         try:
--> 353             obj, end = self.scan_once(s, idx)
    354         except StopIteration as err:
    355             raise JSONDecodeError("Expecting value", s, err.value) from None

JSONDecodeError: Invalid control character at: line 1 column 20 (char 19)

In [2]: import json
   ...: json.loads('{"job_name": "hello\\nworld"}')
Out[2]: {'job_name': 'hello\nworld'}
```

If so that's something the ARC developers should fix as it's explictly forbidden in the [section 9 of the JSON spec](https://www.ecma-international.org/publications-and-standards/standards/ecma-404/):

> A string is a sequence of Unicode code points wrapped with quotation marks (U+0022). All code points may be placed within the quotation marks except for the code points that must be escaped: quotation mark (U+0022), reverse solidus (U+005C), and the control characters U+0000 to U+001F. There are two-character escape sequence representations of some characters.
> * \\" represents the quotation mark character (U+0022).
> * \\\\ represents the reverse solidus character (U+005C).
> * \\/ represents the solidus character (U+002F).
> * \\b represents the backspace character (U+0008).
> * \\f represents the form feed character (U+000C).
> * \\n represents the line feed character (U+000A).
> * \\r represents the carriage return character (U+000D).
> * \\t represents the character tabulation character (U+0009).

Reply:
I will point it out again and get back on this.

Reply:
ARC developers have finally agreed that there is a problem - and fixed it! To come in the next release of ARC software(v6.14) as I understand it. So, hopefully in the next 2 weeks or so. Will remove this hack (and other similar lines) once the fix within ARC is confirmed to work. Keeping open till then.

Reply:
Finally confirmed that it is fixed in the newest ARC v6.14+.